lib/optimizely/config_manager/http_project_config_manager.rb

```diff
@@ -0,0 +1,223 @@
+      )
+      begin
+        headers = {
+          'Content-Type' => 'application/json'
+        }
+
+        headers[Helpers::Constants::HTTP_HEADERS['LAST_MODIFIED']] = @last_modified if @last_modified
+
+        response = HTTParty.get(
+          @datafile_url,
+          headers: headers,
+          timeout: Helpers::Constants::CONFIG_MANAGER['REQUEST_TIMEOUT']
+        )
+      rescue StandardError => e
+        @logger.log(
+          Logger::ERROR,
+          "Fetching datafile from #{@datafile_url} failed. Error: #{e}"
+        )
+        return
+      end
+
+      handle_response response
+    end
+
+    private
+
+    def polling_interval(polling_interval)
+      # Sets frequency at which datafile has to be polled and ProjectConfig updated.
+      #
+      # polling_interval - Time in seconds after which to update datafile.
+
+      # If polling interval is less than minimum allowed interval then set it to default update interval.
+
+      if polling_interval.to_i >= Helpers::Constants::CONFIG_MANAGER['MIN_UPDATE_INTERVAL']
+        @polling_interval = polling_interval
+        return
+      end
+
+      @logger.log(
+        Logger::DEBUG,
+        "Invalid update_interval #{polling_interval} provided. Defaulting to #{Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']}"
+      )
+      @polling_interval = Helpers::Constants::CONFIG_MANAGER['DEFAULT_UPDATE_INTERVAL']
+    end
+
+    def get_datafile_url(sdk_key, url, url_template)
+      # Determines URL from where to fetch the datafile.
+      # sdk_key - Key uniquely identifying the datafile.
+      # url - String representing URL from which to fetch the datafile.
+      # url_template - String representing template which is filled in with
+      #               SDK key to determine URL from which to fetch the datafile.
+      # Returns String representing URL to fetch datafile from.
+
+      raise InvalidInputsError, 'Must provide at least one of sdk_key or url.' if sdk_key.nil? && url.nil?
+
+      unless url
+        url_template ||= Helpers::Constants::CONFIG_MANAGER['DATAFILE_URL_TEMPLATE']
+        begin
+          return (url_template % sdk_key)
+        rescue
+          raise InvalidInputsError, "Invalid url_template #{url_template} provided."
+        end
+      end
+
+      url
+    end
+
+    def handle_response(response)
+      # Helper method to handle response containing datafile.
+      #
+      # response - requests.Response
+
+      # Leave datafile and config unchanged if it has not been modified.
+      if response.code == '304'
+        @logger.log(
+          Logger::DEBUG,
+          "Not updating config as datafile has not updated since #{@last_modified}."
+        )
+        return
+      end
+
+      @last_modified = response[Helpers::Constants::HTTP_HEADERS['LAST_MODIFIED']]
+      config = DatafileProjectConfig.create_project_config_from_datafile(response.body, @logger, @error_handler, @skip_json_validation) if response.body
```

Comment:
I am not sure but @mikeng13 would like to know your thoughts. 
`DatafileProjectConfig` itself representing `Datafile` association with `ProjectConfig`, the method name should be `create` only. :)